lib/AST/ASTPrinter.cpp

```diff
@@ -5134,12 +5158,35 @@ void PrintAST::visitBinaryExpr(BinaryExpr *expr) {
 }
 
 void PrintAST::visitCoerceExpr(CoerceExpr *expr) {
+  visit(expr->getSubExpr());
+  Printer << " as ";
+  printType(expr->getCastType());
 }
 
-void PrintAST::visitOneWayExpr(OneWayExpr *expr) {
-}
+void PrintAST::visitOneWayExpr(OneWayExpr *expr) { }
```

Comment:
Also might be worth adding an `llvm_unreachable` here since we should never see it outside the constraint system